modules/ingester/instance.go

```diff
@@ -239,16 +262,39 @@ func (i *instance) CompleteBlock(blockID uuid.UUID) error {
 		return errors.Wrap(err, "error creating ingester block")
 	}
 
+	// Search data (optional)
 	i.blocksMtx.Lock()
-	i.completeBlocks = append(i.completeBlocks, ingesterBlock)
+	oldSearch := i.searchAppendBlocks[completingBlock]
 	i.blocksMtx.Unlock()
 
+	var newSearch search.SearchBlock
+	if oldSearch != nil {
```

Comment:
seems weird that oldSearch would be nil. should that ever be true? is this an error?

Reply:
Yes it will be nil for any blocks replayed/reloaded on startup that didn't have search data. This can be true for WAL or locally completed blocks from a previous version, or the same version in this case as search is intentionally not reloaded on startup.